### PR TITLE
Setup scripts for Copilot Chat

### PR DESCRIPTION
**Pull request by gitri-ms**
### Motivation and Context

The setup process for Copilot Chat can be complicated for first time users, and we want to increase developer velocity by making the process as simple as possible through automation.

### Description

Add Windows batch scripts for setting up and running Copilot Chat. Per the instructions in the README, these scripts can be used individually to perform each step of the setup process, or the all-in-one scripts can be used for (nearly) one-click deploy.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/scripts/1_install-reqs.cmd [hunk 1]
```diff
@@ -0,0 +1,5 @@
+:: Install chocolatey
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/1_install-reqs.cmd:1]: Please add a comment at the top of each file explaining what it does and why.

File: samples/apps/copilot-chat-app/scripts/1_install-reqs.cmd [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+@"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "[System.Net.ServicePointManager]::SecurityProtocol = 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))" && SET "PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin"
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/1_install-reqs.cmd:2]: All Windows machines have PowerShell these days - have we considered using PowerShell scripts?

File: samples/apps/copilot-chat-app/scripts/3_init-frontend.cmd [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+if "%1" == "" (
+    goto :runapp
+)
+
+:: Create the .env file
+del .\.env
+echo REACT_APP_BACKEND_URI=https://localhost:40443/ >> .\.env
+
+if "%2" == "msa" (
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/3_init-frontend.cmd:13]: Oh now I really think we should use PowerShell. :)

File: samples/apps/copilot-chat-app/scripts/1_install-reqs.cmd [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+:: Install prereqs
+choco install -y dotnet-6.0-sdk nodejs yarn
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/1_install-reqs.cmd:5]: This looks like the only Windows-specific line in your scripts. If we could provide cross-platform support here and use PowerShell 7 (pwsh), we could get cross-platform support for free.

File: samples/apps/copilot-chat-app/scripts/3_init-frontend.cmd [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    echo REACT_APP_AAD_AUTHORITY=https://login.microsoftonline.com/9188040d-6c67-4c5b-b112-36a304b66dad >> .\.env
+) else (
+    if "%2" == "msft" (
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/3_init-frontend.cmd:16]: Do we want to include something that is only specific to internal Microsoft users?

File: samples/apps/copilot-chat-app/scripts/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+These scripts take you through each step of the setup process for running Copilot Chat locally.
+
+### 1. Install prerequisites
+The first script ensures all requirements are installed using the Chocolatey package manager:
+```
+.\1_install-reqs.cmd
+```
+
+### 2. Start the backend
+Before starting the backend, make sure your `appsettings.json` is up to date with the correct endpoints and service specification. You can either store your OpenAI or Azure OpenAI key manually using `dotnet user-secrets`, or let this script do it for you by optionally providing your key as a parameter.
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/README.md:16]: Include a link to a README with instructions for configuration?

File: samples/apps/copilot-chat-app/scripts/start-first-run.cmd [hunk 7]
```diff
@@ -0,0 +1 @@
+.\1_install-reqs.cmd && start cmd.exe /k ".\2_init-backend.cmd %1" & start cmd.exe /k ".\3_init-frontend.cmd %2 %3"
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/start-first-run.cmd:1]: I would consider not having a "first run" specific case. Dependency installation should be idempotent - let's try and keep it simple with just "start.cmd" that either includes "install-reqs" or have the user call install-reqs via instructions in README.

File: samples/apps/copilot-chat-app/scripts/2_init-backend.cmd [hunk 8]
```diff
@@ -0,0 +1,15 @@
+@echo off
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/scripts/2_init-backend.cmd:1]: Both of the "init" scripts actually run the software. Consider removing the numbers and renaming:
- "install-preprequisites.cmd"
- "start-backend.cmd"
- "start-frontend.cmd"
- "start.cmd"

**Comment by adrianwyatt** [comment]
I am *super* excited to see this come in!

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

**Review by dluc — COMMENTED** [review]
could you add also the corresponding Bash scripts please?

**Pull request closed without merging**